Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -418,7 +418,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLRespon
     if (self.statusCode == 401
         && client.accessToken.refreshToken != nil
         && authenticateHeader
-        && [authenticateHeader rangeOfString:@"expired_token"].location != NSNotFound) {
+        && ([authenticateHeader rangeOfString:@"invalid_token"].location != NSNotFound || [authenticateHeader rangeOfString:@"invalid_request"].location != NSNotFound)) {
```

Comment:
Thanks! I'm the iPhone side of this, so I can't speak to the server side code. I'll pass this along to our dev. We were testing against a server that expired the access token every 60 seconds.